zkevm-circuits/src/taiko_pi_circuit.rs

```diff
@@ -540,11 +490,12 @@ impl<F: Field> SubCircuit<F> for TaikoPiCircuit<F> {
     fn unusable_rows() -> usize {
         // No column queried at more than 3 distinct rotations, so returns 6 as
         // minimum unusable rows.
-        6
+        PublicData::<F>::default().total_len() + 3
     }
 
-    fn min_num_rows_block(_block: &witness::Block<F>) -> (usize, usize) {
-        (USED_ROWS, USED_ROWS)
+    fn min_num_rows_block(block: &witness::Block<F>) -> (usize, usize) {
+        // TODO(Cecilia): what is the first field?
```

Comment:
`(rows_without_padding, row_with_padding)`. So for this circuit it should be same number of rows for both.